packages/superset-ui-translation/test/Translator.test.js

```diff
@@ -30,11 +30,48 @@ describe('Translator', () => {
     it('returns undefined for undefined input', () => {
       expect(translator.translate(undefined)).toBeUndefined();
     });
+    it('returns original text for unknown text', () => {
+      expect(translator.translate('abc')).toEqual('abc');
+    });
     it('translates simple text', () => {
       expect(translator.translate('second')).toEqual('秒');
     });
-    it('translates template text with arguments', () => {
+    it('translates template text with an argument', () => {
       expect(translator.translate('Copy of %s', 1)).toEqual('1 的副本');
     });
+    it('translates template text with multiple arguments', () => {
+      expect(translator.translate('test %d %d', 1, 2)).toEqual('test 1 2');
+    });
+  });
+  describe('.translateWithNumber(singular, plural, num, ...args)', () => {
+    const translator = new Translator({
+      languagePack: languagePackZh,
+    });
+    it('returns null if singular or plural is null', () => {
+      expect(translator.translateWithNumber(null, 'plural')).toBeNull();
+      expect(translator.translateWithNumber('singular', null)).toBeNull();
+      expect(translator.translateWithNumber(null, null)).toBeNull();
+    });
+    it('returns undefined if singular or plural is undefined', () => {
+      expect(translator.translateWithNumber(undefined, 'plural')).toBeUndefined();
+      expect(translator.translateWithNumber('singular', undefined)).toBeUndefined();
+      expect(translator.translateWithNumber(undefined, undefined)).toBeUndefined();
+    });
+    it('returns original text for unknown text', () => {
+      expect(translator.translateWithNumber('fish', 'fishes', 1)).toEqual('fish');
```

Comment:
don't need another test, but curious how it behaves with 0 since that is the default `num`. is that singular?

Reply:
I think that depends on how the languagePack is setup. I saw there are DSL-ish thing in the language packs that defines `n` for singular and plural.